cgo/addresses.go

```diff
@@ -81,7 +82,32 @@ func signMessage(cName, cMessage, cAddress, cPassword *C.char) *C.char {
 
 	sEnc := base64.StdEncoding.EncodeToString(sig)
 
-	return successCResponse(sEnc)
+	return successCResponse("%s", sEnc)
+}
+
+//export verifyMessage
+func verifyMessage(cName, cMessage, cAddress, cSig *C.char) *C.char {
+	w, ok := loadedWallet(cName)
+	if !ok {
+		return errCResponse("wallet with name %q is not loaded", goString(cName))
+	}
+
+	addr, err := stdaddr.DecodeAddress(goString(cAddress), w.MainWallet().ChainParams())
+	if err != nil {
+		return errCResponse("unable to decode address: %v", err)
+	}
+
+	sig, err := base64.StdEncoding.DecodeString(goString(cSig))
+	if err != nil {
+		return errCResponse("unable to decode signature: %v", err)
+	}
+
+	ok, err = dcrwallet.VerifyMessage(goString(cMessage), addr, sig, w.MainWallet().ChainParams())
```

Comment:
// Addresses must have an associated secp256k1 private key and therefore
	// must be P2PK or P2PKH (P2SH is not allowed).
	switch addr.(type) {
	case *stdaddr.AddressPubKeyEcdsaSecp256k1V0:
	case *stdaddr.AddressPubKeyHashEcdsaSecp256k1V0:
		// Valid address types, proceed with verification.
	default:
		return errCResponse("invalid address type: must be P2PK or P2PKH")
	}
	
	shouldn't we be checking if the address is P2SH?

Reply:
Oh yeah. I assumed it would error somewhere along the way anyhow but we can check.